### ISSUE TITLE
Animation Player as root node of a scene + adding it into another scene but renaming it breaks it

### ISSUE THREAD

**Issue by FalconX4** [godotengine/godot#92343]
### Tested versions

- Reproducible in: 4.2.2 stable

### System information

Godot v4.2.2.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 750 Ti (NVIDIA; 31.0.15.3742) - Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz (8 Threads)

### Issue description

Creating an animation player as the root of a scene and adding it into another scene but renamed will create an error of get_node: Node not found.

If its already fixed in other version nice otherwise a possible way to fix this would be to change this:
![image](https://github.com/godotengine/godot/assets/1879422/4f30ade4-45f0-4933-af71-5e37723aec35)
Into this:
![image](https://github.com/godotengine/godot/assets/1879422/91d1b764-9a20-43d1-a511-01eee6953ee7)

So having the node path start with "." when the animation player is the root of the scene and adding "root_node = NodePath("./")" under the node definition. After doing this somehow, it stays like that even after resaving.

### Steps to reproduce

1. Create a new scene using other node -> animation player
2. Create an animation and set it to autoplay on load
3. Save it
4. Create another scene
5. Add the scene create at 1 and rename it
6. Load the scene

### Minimal reproduction project (MRP)

Not project dependent but here it is anyway

[AnimationPlayerRootNodeBug.zip](https://github.com/godotengine/godot/files/15440542/AnimationPlayerRootNodeBug.zip)

**Comment by theraot** [comment]
This changed from an error to a warning in v4.3.dev4.official [df78c0636], aside form that the issue is still present in v4.3.dev6.official [89850d553] and v4.3.beta.custom_build [a4f2ea91a]

For reference for future readers, I'll reiterate the issue: The `AnimationPlayer` has an animation with an animation track that references itself, and the generated `NodePath` for said track has the name of the `AnimationPlayer` instead of using `.`, As a result when renaming the `AnimationPlayer` the animation breaks.

**Comment by fire** [comment]
This is an error.

However I am trying to figure out why you want do this.

~~Since animation is a node it doesn't have a global transform and recreating the global transform is as much effort as implementing the node3d~~

The animation player generally calls the subnodes it has by convention, so if you want to animate the animation player you need another animation player parent to edit it.

By design editing the animation player node itself and editing the animation played by an animation is a different operation.

tl;dr I don't get it. Will look at your description more.
